core/backend/src/test/standalone/GeometryStream.test.ts

```diff
@@ -2476,6 +2466,53 @@ describe("BRepGeometry", () => {
       assert(false, error.message);
     }
   });
+
+  it("catch solid kernel severe error - main thread", async () => {
+    const builder = new ElementGeometry.Builder();
+    builder.appendGeometryQuery(Loop.createPolygon([Point3d.create(0, 0, 0), Point3d.create(0, 1, 0), Point3d.create(-1.1, 1, 0), Point3d.create(-1, 1.1, 0)]));
+
+    // Test main thread using createBRepGeometry...
+    const onResult: BRepGeometryFunction = (_info: BRepGeometryInfo): void => { };
+
+    const createProps: BRepGeometryCreate = {
+      operation: BRepGeometryOperation.Offset,
+      entryArray: builder.entries,
+      onResult,
+      parameters: { distance: 0.25 },
+    };
+    try {
+      imodel.createBRepGeometry(createProps);
+      assert(false); // Expect exception creating sheet body from invalid loop...
+    } catch (error) {
+      assert(true);
+    }
+  });
+
+  it("catch solid kernel severe error - worker thread", async () => {
+    const builder = new ElementGeometry.Builder();
+    const loop0 = Loop.createPolygon([Point3d.create(0, 0, 0), Point3d.create(0, 1, 0), Point3d.create(-1.1, 1, 0), Point3d.create(-1, 1.1, 0)]);
+    const loop1 = Loop.create(Arc3d.createXY(Point3d.create(0, 0, 1), 0.2));
+    const solid = RuledSweep.create([loop0, loop1], true);
+    assert.isDefined(solid);
+    builder.appendGeometryQuery(solid!);
+
+    // Test worker thread using getMassProperties...
+    const result = createGeometricElemFromSeed(imodel, "0x1d", builder.entries);
+    assert.isTrue(DbResult.BE_SQLITE_OK === result.status);
+
+    const requestProps: MassPropertiesRequestProps = {
+      operation: MassPropertiesOperation.AccumulateVolumes,
+      candidates: [result.newId],
+    };
+
+    const requestContext = new BackendRequestContext();
+    try {

```

Comment:
`expect(imodel.getMassProperties(requestContext, requestProps)).to.be.rejectedWith(...)`

Reply:
Pushed.